client/app/widgets/ToggleLocale/__tests__/ToggleLocale.spec.jsx

```diff
@@ -17,23 +22,25 @@ describe('ToggleLocale', () => {
   });
 
   it('does nothing if the previous locale is the same as the selected', () => {
-    const wrapper = mount(component);
+    render(component);
     act(() => {
-      wrapper.find('select').prop('onChange')({
-        currentTarget: { value: 'en' },
+      fireEvent.select(screen.getByRole('combobox'), {
+        target: { value: 'en' },
       });
     });
     expect(axios.post).not.toHaveBeenCalled();
   });
 
-  it('sets the locale cookie and makes a post request if the selected locale is different from the previous', () => {
-    const wrapper = mount(component);
+  it('sets the locale cookie and makes a post request if the selected locale is different from the previous', async () => {
+    render(component);
     act(() => {
-      wrapper.find('select').prop('onChange')({
-        currentTarget: { value: 'es' },
+      fireEvent.change(screen.getByRole('combobox'), {
+        target: { value: 'es' },
       });
     });
     expect(Cookies.set).toHaveBeenCalledWith('locale', 'es');
-    expect(axios.post).toHaveBeenCalledWith('/toggle_locale', { locale: 'es' });
+    await waitFor(() => expect(axios.post).toHaveBeenCalledWith('/toggle_locale', {
```

Comment:
Since this test is only checking if the request gets triggered, it is not necessary to `await` the response. The request does not need to conclude, as the response itself is not used here. We can simply expect `axios.post` to have been called (even if it concludes async). For now, I think this assertion can be kept synchronous like it was before 👍

Reply:
Makes sense. Thanks!